openshift-kube-apiserver/admission/customresourcevalidation/route/validation_wrapper.go

```diff
@@ -0,0 +1,102 @@
+package route
+
+import (
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/apiserver/admission/admissionrestconfig"
+	routecommon "github.com/openshift/library-go/pkg/route"
+	authorizationv1client "k8s.io/client-go/kubernetes/typed/authorization/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/component-base/featuregate"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
+)
+
+type validateCustomResourceWithClient struct {
+	admission.ValidationInterface
+
+	secretGetter        corev1client.SecretsGetter
+	sarc                authorizationv1client.SubjectAccessReviewInterface
+	routeValidationOpts routecommon.RouteValidationOptions
+}
+
+func NewValidateRoute() (admission.Interface, error) {
```

Comment:
could you please add a unit test that would ensue that the `getters` can be injected?